vulture_os/toolkit/api_parser/hornetsecurity/hornetsecurity.py

```diff
@@ -0,0 +1,361 @@
+                logger.exception(f"[{__parser__}][execute_query]: Unexpected error while fetching {kind}'s logs - {since, to} -- {e}", extra={'frontend': str(self.frontend)})
+                retry -= 1
+                continue
+
+            try:
+                json_resp = resp.json()
+            except Exception as e:
+                logger.exception(f"[{__parser__}][execute_query]: Failed to decode json response {resp.content} -- {e}", extra={'frontend': str(self.frontend)})
+                raise HornetSecurityAPIError(e)
+
+            # Handles more than 10k logs to retrieve in a single query
+            toRetrieve = json_resp.get('num_found_items', 0)
+            if toRetrieve >= 10000: # error_id returned by API in this case is 3089
+                return "divide", toRetrieve, []
+
+            if resp.status_code != 200:
+                logger.exception(f"[{__parser__}][execute_query]: Receive {resp.status_code} != 200 while fetching {kind}'s logs - {since, to} -- {resp.content}", extra={'frontend': str(self.frontend)})
+                retry -= 1
+                continue
+
+            if kind == "auditing":
+                return "write", toRetrieve, json_resp.get('auditings', [])
+            elif kind == "emails":
+                return "write", toRetrieve, json_resp.get('emails', [])
+
+        msg = f"Fail to fetch {kind} by encountering more than 3 errors consecutively {since, to}"
+        logger.exception(f"[{__parser__}][execute_query]: {msg}", extra={'frontend': str(self.frontend)})
+        raise HornetSecurityAPIError(msg)
+
+
+    def get_logs(self, kind: str, since: datetime, to: datetime) -> tuple[list, datetime]:
+        """
+        Get logs by kind specification and (since; to) couple
+        If there is more than 10000 logs to retrieve, the collector continuously divide the range of time to get by two; reducing probability to encounters more than the limit number of logs
+            -> there is no possibility to query with a request containing an (offset + limit) >= 10 000
+            -> there is no possibility to sort
+            -> by default logs are timestamp descendently sorted (thus we always retrieve the max serie timestamp on the first chunk's log)
+        """
+        # API doesn't supports microseconds precision we must remove it
+        since = since.replace(microsecond=0)
+        to = to.replace(microsecond=0)
+
+        ret = []
+
+        limit = 1000
+        op, count, logs = self.execute_query(kind, since, to, limit, len(ret))
+        if op == "divide":
+            to = since + ((to - since) / 2)
+            logger.info(f"[{__parser__}][get_logs]: We've tried to retrieve more than 10k {kind} in one query ({count}) - dividing the range of time by 2 -> {since.isoformat(), to.isoformat()}", extra={'frontend': str(self.frontend)})
+        elif op == "write":
+            ret.extend(logs)
+
+        while (len(ret) != count and not self.evt_stop.is_set()):
```

Comment:
prefer doing 
```suggestion
        while (len(ret) < count and not self.evt_stop.is_set()):
```
to avoid conditions where we receive more logs than expected (even if that may seem unlikely)